### DISCUSSION TITLE
Multiple inputs and outputs.

### DISCUSSION THREAD

**Discussion by AtalibaJunior** [lathoub/Arduino-AppleMIDI-Library#115] [Q&A]
Is it possible to create more than 1 virtual cable, for example, (3 in and 3 outs)? How does the computer's operating system recognize these cables through a single IP?

**Comment by lathoub** [comment]
RTP-MIDI (aka AppleMIDI) uses a concept of [sessions](https://en.wikipedia.org/wiki/RTP-MIDI#Sessions). 

(I'm a little confused by the use of the word 'cable', and feels like a [USB-MIDI concept](https://github.com/lathoub/Arduino-USBMIDI)).

If you are on [Windows](https://en.wikipedia.org/wiki/RTP-MIDI#Windows), install the rip-midi driver and then have your Arduino to connect to it and start to play around. (MacOS has built in support for RTP-MIDI)
